### ISSUE TITLE
Screen cheese when reloading with a notebook in a floating window

### ISSUE THREAD

**Issue by bpasero** [microsoft/vscode#201832]
Steps to Reproduce:

1. move a notebook into a aux window
2. reload window

🐛 screen cheese restoring the notebook

<img width="1612" alt="image" src="https://github.com/microsoft/vscode/assets/900690/bbbaab53-e67f-4e13-9a6c-97cb2f81797f">

**Comment by rebornix** [comment]
@bpasero thanks for the report. I found that I'm struggling with the inconsistency of `activeContainer` or `codeWindow`:

* If I use `this.layoutService.activeContainer` all the time
  * it works for opening notebook in main window
  * it works when opening in floating window
  * when closing the floating window, the notebook editor will never get restored properly as `activeContainer` is still being set to the floating window, but it should be the main window when we try to create the notebook again.
* Thus I started to check what's the owner code window for the notebook editor container DOM node by `DOM.getWindow(this._rootElement)`, it fixes all above scenarios.

However this approach breaks when reloading VS Code. Digging into this I found that after window reload and we attempt to restore the notebook editor, `DOM.getWindow(this._rootElement)` always returns the main code window instead of the floating one.

https://github.com/microsoft/vscode/blob/6079e24015ed3594788b1e68f1d5af764cdd9003/src/vs/workbench/contrib/notebook/browser/notebookEditor.ts#L209

`this._rootElement`'s `ownerDocument` turns out to be the main window's document. I guess it's because `this._rootElement` is created under an `editor-instance` div that's created under the main document.

With the inconsistency, I couldn't figure out where to attach the notebook DOM nodes always correctly. @bpasero is this an oversight?

**Comment by bpasero** [comment]
Interesting, we do begin to restore floating windows from here:

https://github.com/microsoft/vscode/blob/d2028ff9c7678834f810cad53f10a35718f3fcba/src/vs/workbench/browser/parts/editor/editorParts.ts#L211-L216

Specifically creating a floating window per state:

https://github.com/microsoft/vscode/blob/d2028ff9c7678834f810cad53f10a35718f3fcba/src/vs/workbench/browser/parts/editor/editorParts.ts#L280-L284

The editor state is passed into the editor part of the floating window here:

https://github.com/microsoft/vscode/blob/d2028ff9c7678834f810cad53f10a35718f3fcba/src/vs/workbench/browser/parts/editor/auxiliaryEditorPart.ts#L103

And then we create the grid widget with that state:

https://github.com/microsoft/vscode/blob/d2028ff9c7678834f810cad53f10a35718f3fcba/src/vs/workbench/browser/parts/editor/editorPart.ts#L1173

However we only seem to be setting the grid widget to its parent container after restoring state:

https://github.com/microsoft/vscode/blob/d2028ff9c7678834f810cad53f10a35718f3fcba/src/vs/workbench/browser/parts/editor/editorPart.ts#L1246

So as long as that did not happen, the owner document of the grid widget will be the main window!

This is an issue because we decided to run `document.createElement` always from the main window document and not the floating window document, otherwise the changes in our code would be a magnitude more involved.

**tl;dr**
* until a `HTMLElement` is appended to its parent in the floating window, its owner document is the main window
* we always need to ensure to first append and then judge about the owner document
* this is hard to enforce and tbh I am expecting a lot more issues like these that we need to iron out going forward

**Comment by rebornix** [comment]
@bpasero thanks for sharing the details. I found we kind of have the right info to find the right code window for an editor input/pane, it's not yet exposed, so maybe we could introduce an API which everyone can use.

Here is my workaround to get this issue resolved, it's not great but I still merge this to ensure it's working, https://github.com/microsoft/vscode/pull/201904, what I was trying to do:

* The editor input can access its group, I should ideally use `this._editorGroupService.getPart(this.group.id)` to get its part. However it seems during window reload, the aux window has zero groups. For example, if the main window has group 1 and 2, aux window has group 3, during window reload, main window does restore group 1 and 2, but it seems aux window has zero groups. So calling `this._editorGroupService.getPart(this.group.id)` can't find a matching part, and it falls back to main window.
  * I work around this via https://github.com/microsoft/vscode/pull/201904/files#diff-330f92e5612e8f9dcc91e85d694b6e10fdaa5f14c384157edfd27dfbbf0f8684R214-R215
* Once I get the right part, I attempt to read `windowId` so I can use that to revive the `codewindow`.

It's now working fine for all scenarios, but it would be awesome that there is one unified accurate API to get what's the right `container` to use for an editor input.

**Comment by bpasero** [comment]
I will see to allow to access `group.windowId` to make this easier. However I think the real fix would be that the grid widget first appends created DOM nodes to the parent before going down the path of handing it off to components such as the editor. I will ping Joao on this and maybe we leave that part up for debt week as it may have some consequences.
